### PR TITLE
CE-624: CKEditor link tool allows invalid URLs

### PR DESCRIPTION
**Pull request by RyanCopley**
### Before
screen shot 2018-06-04 at 10 08 02 am

### After

screen shot 2018-06-04 at 10 08 02 am

### Video
jun-04-2018 11-24-08

PLEASE REVIEW INDIVIDUAL COMMITS

### CODE REVIEW

File: plugins/link/dialogs/link.js [hunk 1]
```diff
@@ -94,76 +94,53 @@
...
+                                        this.allowOnChange = !0
+                                    },
+                                    onKeyUp: function() {
+                                    },
+                                    onChange: function() {
+                                    },
+                                    validate: function() {
+                                        var a = this.getDialog();
+                                        try {
+                                            new URL(this.getValue());
```
Comment (RyanCopley) [inline plugins/link/dialogs/link.js]: this is the majority of where the adjustments are

File: plugins/link/dialogs/link.js [hunk 2]
```diff
@@ -94,76 +94,53 @@
                 }, {
                     type: "vbox",
                     id: "urlOptions",
-                    children: [{
-                        type: "hbox",
-                        widths: ["25%", "75%"],
-                        children: [{
-                            id: "protocol",
```
Comment (RyanCopley) [inline plugins/link/dialogs/link.js]: I talked with Kate Chipman about removing the "protocol" dialog and allowing the users to just paste in the URL into 1 box which is why this is removed

**Review by vpratti — APPROVED** [review]

**Review by jhaugh42 — APPROVED** [review]
I'd ask for tests.  Are there any tests for this?

**Comment by jhaugh42** [comment]
Don't break everything please. :)

**Pull request merged**